### DISCUSSION TITLE
Links other than messaging on Graph tab possible?

### DISCUSSION THREAD

**Discussion by milcent-CVM** [sepinf-inc/IPED#2067] [Q&A]
Is there a way of "tricking" the links graph to add nodes and edges other than messaging ones?
For example, if I'm indexing documents that cite other documents, like scientific papers or court decisions, and I already have a citation table, would it be possible to send these relationships to the graph?
If so, would it have to be at indexing time or could it be done afterwards?
Thanks in advance!

**Comment by lfcnassif** [comment]
Unfortunately not. The graph is populated at indexing time using the `Message:From` and `Message:To` properties of the decoded items.

**Reply by milcent-CVM** [reply]
If there was a way of injecting `Message:From` and `Message:To` as metadata to the files, could it work?

**Reply by lfcnassif** [reply]
Yes, although those properties are intended for communication relations, it should work. You can populate those Metadata properties into the desired items/documents creating a new script task and installing it into conf/TaskInstaller.xml.

**Reply by milcent-CVM** [reply]
Great! Thank you!!
Would it have to be in java only, or is there a chance I could implment it in Python? :)

**Reply by lfcnassif** [reply]
Yes, it can be implemented in python.

**Reply by lfcnassif** [reply]
An important thing to keep in mind about the way IPED's graph was designed:
- `Message:To` and `Message:From` values are shown as nodes
- Case items are shown as edges (e.g. emails and instant messages)

So when you select edges, the related items are filtered in the results table.

**Reply by lfcnassif** [reply]
There is an option in conf/GraphConfig.json to plot Regex results found into documents (like CPFs, CNPJs, emails, URLs, phones and other entities) in the graph as nodes and the containing documents as edges connecting every entity pair found into the same document (if 2 entities are present into the same doc, they may be related, that was the theory in the past...). But it is deprecated and probably will be removed in the future, since I don't agree 100% with that theory and it results in a quadratic number of edges to the number of entities found, making the graph bloated and stuck in some situations.
